src/sentry/static/sentry/app/views/settings/components/settingsLayout.tsx

```diff
@@ -14,18 +19,86 @@ type Props = {
   children: React.ReactNode;
 } & RouteComponentProps<{}, {}>;
 
-function SettingsLayout(props: Props) {
-  const {params, routes, route, router, renderNavigation, children} = props;
+type State = {
+  navVisible: boolean;
+  navOffsetTop: number;
+};
+
+class SettingsLayout extends React.Component<Props, State> {
+  static propTypes = {
+    renderNavigation: PropTypes.func,
+    route: PropTypes.object,
+    router: PropTypes.object,
+    routes: PropTypes.array,
+  };
+
+  state = {
+    /**
+     * This is used when the screen is small enough that the navigation should
+     * be hidden. On large screens this state will end up unused.
+     */
+    navVisible: false,
+    /**
+     * Offset mobile settings navigation by the height of main navigation,
+     * settings breadcrumbs and optional warnings.
+     */
```

Comment:
Nit: If these comments were on the type then they would show up in autocomplete.